### ISSUE TITLE
[Bug]: nextcloud 30 RC1 - Filter does not survive the change to a subdirectory

### ISSUE THREAD

**Issue by devnull4242** [nextcloud/server#47251]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I set the filter e.g. to "folder" and "PDFs" and hope that i can navigate to subfolders to find only PDFs.

### Steps to reproduce

1. Nextcloud 30 RC 1
2. set filter to e.g. "folder" and "PDFs"
3. filter is set
4. change to subfolder
5. filter is unset

### Expected behavior

After change to subfolder the filter is not unset but set to e.g. "folder" and "PDFs".

![2024-08-15 07_34_59-Window](https://github.com/user-attachments/assets/f78a0531-99ce-4181-8402-43a7c87a62a3)

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

master

### Operating system

Other

### PHP engine version

None

### Web server

Other

### Database engine version

Other

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 28 to 29)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by susnux** [comment]
This is done by design. 

CC @sorbaugh @jancborchardt should we maybe only reset when changing the view?

**Comment by sorbaugh** [comment]
Resetting filters only when changing the view sounds good to me. @susnux would there be any potential pitfalls? But yeah, waiting also for @jancborchardt 's input.

**Comment by susnux** [comment]
> would there be any potential pitfalls?

We had bug reports in the past that wanted to reset the search on navigation. Thats why we reset the filters when changing the directory.
But now with the filters it might make sense to keep it active e.g. if you want to check also sub-directories for your search.

**Comment by jancborchardt** [comment]
Yes, agree with the bug report and to keep the filter active when doing simple folder navigation.
